### PR TITLE
test: improve reliability of canary rgw-multisite-testing

### PR DESCRIPTION
**Pull request by jhoblitt**
Related to #14841

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by jhoblitt** [comment]
Add a wait on CephObjectStore to be ready rather than the old logic if waiting for the pods has shown that the problem is that rgw never becomes ready.

```
+ timeout 300 bash
++ kubectl -n rook-ceph get cephobjectstore multisite-store -o 'jsonpath={..status.phase}'
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
waiting for CephObjectStore/multisite-store in rook-ceph to be Ready
Error: Process completed with exit code 124.
```

**Comment by jhoblitt** [comment]
It seems that some times it takes a while for rgw to come up.

```
Events:
  Type     Reason              Age                    From                         Message
  ----     ------              ----                   ----                         -------
  Warning  ReconcileFailed     7m6s (x10 over 8m6s)   rook-ceph-object-controller  failed to reconcile CephObjectStore "rook-ceph/multisite-store". failed to create object store deployments: failed to configure multisite for object store: failed join ceph multisite in zone "zone-a": failed to get rgw zone: exit status 2
  Warning  ReconcileFailed     6m29s (x3 over 6m56s)  rook-ceph-object-controller  failed to reconcile CephObjectStore "rook-ceph/multisite-store". failed to create object store deployments: failed to get COSI user "cosi": Get "http://rook-ceph-rgw-multisite-store.rook-ceph.svc:80/admin/user?format=json&uid=cosi": dial tcp 10.107.116.158:80: connect: connection refused
  Normal   ReconcileSucceeded  6m4s (x10 over 9m41s)  rook-ceph-object-controller  successfully configured CephObjectStore "rook-ceph/multisite-store"
```

I had to work 4 times in a row earlier today... my guessing that it will eventually always come up if we wait long enough.

**Comment by jhoblitt** [comment]
5 passes in a row now.

File: .github/workflows/canary-integration-test.yml [hunk 1]
```diff
@@ -1528,8 +1528,75 @@ jobs:
       - name: set Ceph version in CephCluster manifest
         run: tests/scripts/github-action-helper.sh replace_ceph_image  "deploy/examples/cluster-test.yaml" "${{ github.event.inputs.ceph-image }}"
 
-      - name: run RGW multisite test
-        uses: ./.github/workflows/rgw-multisite-test
+      - name: setup cluster resources
```
Comment (travisn) [inline .github/workflows/canary-integration-test.yml:1531]: Did anything change in this script, or just moved from the other file?

**Review by travisn — COMMENTED** [review]
LGTM to stabilize the tests, thanks for improving this! Just a small question...

File: .github/workflows/canary-integration-test.yml (hunk 1 above)
Comment (jhoblitt) [reply .github/workflows/canary-integration-test.yml:1531]: There were some changes.  The inlinification is in separate commit to try to make it easy to view the changes to the workflow. https://github.com/rook/rook/pull/14842/commits/92d9f994c2bdced5b4bebed8790be8c1c49b53c0

File: tests/scripts/github-action-helper.sh [hunk 2]
```diff
@@ -561,7 +566,7 @@ function write_object_read_from_replica_cluster() {
   # a direct sub-shell.
   S3CMD_ERROR=0
   (
-    sleep 60
+    sleep 300
```
Comment (BlaineEXE) [inline tests/scripts/github-action-helper.sh:569]: Is this the effective change that helps stabilize the test? 😅

**Review by travisn — APPROVED** [review]

File: tests/scripts/github-action-helper.sh (hunk 2 above)
Comment (jhoblitt) [reply tests/scripts/github-action-helper.sh:569]: Yes/no/maybe?  It was initially extended to 600s, which seemed to work. I am hopeful that the new waits on the `cephcluster` and `cephobjectstore` CRs would make it unnecessary to wait so long for object replication to work. I did see a failure when trying to change it back to 60s.  My thinking is we would set it a little long and then try to tighten it up.

**Review by BlaineEXE — APPROVED** [review]

File: tests/scripts/github-action-helper.sh (hunk 2 above)
Comment (BlaineEXE) [reply tests/scripts/github-action-helper.sh:569]: Got it! I'm glad to have any improvement in stability here. It's been a long time needed.

Comment (jhoblitt) [reply tests/scripts/github-action-helper.sh:569]: On the most recent run in this PR, it took 94s.  I suspect there is significant jitter.

**Pull request merged**